hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2227,6 +2231,11 @@ func reconcileCAPIManagerClusterRole(role *rbacv1.ClusterRole) error {
 			Resources: []string{"customresourcedefinitions"},
 			Verbs:     []string{"get", "list", "watch"},
 		},
+		{
+			APIGroups: []string{"capi-provider.agent-install.openshift.io"},
```

Comment:
I'd expect this to be scoped to a particular namespace as in `reconcileCAPIManagerRole`

Reply:
I don't think the core CAPI controller needs to know anything about `capi-provider.agent-install.openshift.io`

Reply:
Fixed